script/releases/v1.0.0-slashing/1-deployContracts.s.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import {EOADeployer} from "zeus-templates/templates/EOADeployer.sol";
+import "../Env.sol";
+
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+
+/**
+ * Purpose: use an EOA to deploy all of the new contracts for this upgrade. 
+ */
+contract Deploy is EOADeployer {
+    using Env for *;
+
+    function _runAsEOA() internal override {
+        vm.startBroadcast();
+
+        /// permissions/
+
+        address[] memory pausers = new address[](3);
+        pausers[0] = Env.pauserMultisig();
+        pausers[1] = Env.opsMultisig();
+        pausers[2] = Env.executorMultisig();
+
+        deployImpl({
+            name: type(PauserRegistry).name,
+            deployedTo: address(new PauserRegistry({
+                _pausers: pausers,
+                _unpauser: Env.executorMultisig()
+            }))
+        });
+
+        deployImpl({
+            name: type(PermissionController).name,
+            deployedTo: address(new PermissionController())
+        });
+
+        deployProxy({
+            name: type(PermissionController).name,
+            deployedTo: address(new TransparentUpgradeableProxy({
+                _logic: address(Env.impl.permissionController()),
+                admin_: Env.proxyAdmin(),
+                _data: ""
+            }))
+        });
+
+        /// core/
+
+        deployImpl({
+            name: type(AllocationManager).name,
+            deployedTo: address(new AllocationManager({
+                _delegation: Env.proxy.delegationManager(),
+                _pauserRegistry: Env.impl.pauserRegistry(),
+                _permissionController: Env.proxy.permissionController(),
```

Comment:
How can we validate in Zeus this contract is correctly deployed and not some 0 address? Because this is being deployed from above from the `deployImpl` and `deployProxy` calls

Reply:
- You can check the deployed address, via the `deployments()` [struct](https://github.com/Layr-Labs/zeus-templates/blob/master/src/templates/EOADeployer.sol#L76)
- You can check the updated tracked value within zeus via the `Env` object (see how it fetches the contract address in there). In our tests, we typically grab the before and after addresses, assert that there was a change, and then test certain things about the deployed contract. (This is a forge fork test).

Reply:
Check the tests in this file now - they thoroughly test the newly-deployed contracts